server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/date/ExtendedDateTimeUnit.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.utils.date;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * Enum representing the extended date time units supported for star tree index as part of index mapping.
+ * The enum values are:
+ * <ul>
+ *     <li>HALF_HOUR_OF_DAY: Represents half hour of day rounding</li>
+ *     <li>QUARTER_HOUR_OF_DAY: Represents quarter hour of day rounding</li>
+ * </ul>
+ * <p>
+ * The enum also provides a static map of date field units to their corresponding ExtendedDateTimeUnit instances.
+ *
+ * @see org.opensearch.common.Rounding.DateTimeUnit for more information on the dateTimeUnit enum and rounding logic.
+ *
+ * @opensearch.experimental
+ */
+public enum ExtendedDateTimeUnit implements DateTimeUnitRounding {
+    HALF_HOUR_OF_DAY("half-hour") {
+        @Override
+        public long roundFloor(long utcMillis) {
+            return utcMillis - (utcMillis % TimeUnit.MINUTES.toMillis(30));
+        }
+    },
+    QUARTER_HOUR_OF_DAY("quarter-hour") {
+        @Override
+        public long roundFloor(long utcMillis) {
+            return utcMillis - (utcMillis % TimeUnit.MINUTES.toMillis(15));
+        }
+    };
+
```

Comment:
Can't this be supported for existing OpenSearch aggs framework, and not just be specific for star tree?

Reply:
Yeah the issue is , the current framework is tied to TemporalField , so I tried finding half hour / quarter hour in its implementations and I'm not able to find one. 

The other way is to change the interfaces but between usage of fixed interval where user can specify any minutes and existing calendar intervals being compatible with all timezones, not sure if there is any relevant use case.